src/schema/custom-scalar.ts

```diff
@@ -0,0 +1,28 @@
+import { GraphQLScalarType } from 'graphql';
+
+export const StringOrArray = new GraphQLScalarType({
+	name: 'StringOrArray',
+	description: 'can be a string Or array',
+	serialize(value) {
+		if (typeof value !== 'string' && typeof value !== 'object') {
+			throw new Error('Value must be either a String or an Array');
+		}
+
+		if (typeof value === 'object' && typeof value.length !== 'number') {
+			throw new Error('Value must be either a String or an Array');
+		}
+
+		return value;
```

Comment:
can we write it like this  : 
`if (typeof value !== 'string' && typeof value !== 'object' || typeof value === 'object' && typeof value.length !== 'number') `